core/src/main/scala/org/apache/spark/sql/execution/columnar/ExternalStoreUtils.scala

```diff
@@ -410,29 +410,23 @@ object ExternalStoreUtils {
       cachedBatches: Iterator[CachedBatch],
       requestedColumns: Array[String],
       schema: StructType): Iterator[InternalRow] = {
-    /* TODO: (native code gen does not work for some reason)
     // check and compare with InMemoryColumnarTableScan
     val numColumns = requestedColumns.length
     val columnIndices = new Array[Int](numColumns)
     val columnDataTypes = new Array[DataType](numColumns)
     for (index <- 0 until numColumns) {
       val fieldIndex = schema.fieldIndex(requestedColumns(index))
-      columnIndices(index) = fieldIndex
+      columnIndices(index) = index
```

Comment:
This is because , we already have a pruned CachedBatches while selecting from gem layer.